pkg/proxy/kafka.go

```diff
@@ -332,20 +346,21 @@ func (k *kafkaRedirect) handleResponses(done <-chan struct{}, pair *connectionPa
 		default:
 		}
 
-		// FIXME: reqRecord contains the log record of the first
-		// request, this is not valid information for subsequent
-		// responses
-		record := k.newLogRecordFromResponse(rsp, reqRecord)
-
 		if err != nil {
+			record := k.newLogRecordFromResponse(nil, nil)
 			record.log(accesslog.VerdictError,
 				kafka.ErrInvalidMessage,
 				fmt.Sprintf("Unable to parse Kafka response: %s", err))
 			scopedLog.WithError(err).Error("Unable to parse Kafka response; closing Kafka response connection")
 			return
 		}
 
+		// Find the request that correlates with this response based on the
+		// correlation ID
```

Comment:
I am not sure if `correlation ID` is enough to say that a particular response belongs to a request. `correlation ID` is a user specified field. So 2 requests with different topics can have the same correlation ID back to back. We might not do the correct thing in that scenario. Would it make sense to have both correlation ID as well as topic /client ID to correlate request as well as response?

Reply:
Good point. We can also rely on order + correlation id. What worries me is the following in the spec:

> All requests are initiated by the client, and result in a corresponding response message from the server except where noted.

I haven't found any details in the spec on when exactly that is the case.

Reply:
Discussed more with @raybejjani. He brought up the idea that we could modify the correlationID and set it back in the response direction. This seems the most reliable way to go forward so I will implement this.